config/hooks/pre-commit

```diff
@@ -1,10 +1,17 @@
-#!/bin/sh -e
+#!/bin/bash -ex
 
 REPO_ROOT_DIR="$(git rev-parse --show-toplevel)"
 
-files=$( (git diff --cached --name-only --diff-filter=ACMR | grep -Ei '\.java$') || true)
-if [ -n "$files" ]; then
-    comma_files=$(echo "$files" | paste -s -d , -)
-    "${REPO_ROOT_DIR}/gradlew" goJF -DgoogleJavaFormat.include="$comma_files" >/dev/null 2>&1
-    git add $(echo "$files" | paste -s -d ' ' -)
+mapfile -t files < <(git diff --cached --name-only --diff-filter=ACMR | grep -Ei '\.java$')
```

Comment:
Add a comment on what this command does?  I've seen many bash scripts but I didn't know about `mapfile`

Reply:
Added:
https://github.com/wala/WALA/blob/31eef06302f59940f743858a5891302c1ade585b/config/hooks/pre-commit#L5-L6

Reply:
We still run with `-ex` but we don't have an `|| true` with the new `mapfile` command.  Will the script still succeed even if the `grep` command fails?

Reply:
Also, on Mac:
```shell
$ /bin/bash --version
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin18)
Copyright (C) 2007 Free Software Foundation, Inc.
```
So I'm not sure the `mapfile` command will work.

Reply:
Even if `pipefail` were turned on (which it is not), failure in the process-substitution subcommand would not propagate into an error status in the main flow of the script, and therefore would not terminate the script under `set -e`.

But yes, the archaic Bash version on macOS is going to be a problem. I will need to use an alternate approach. Drat.

Reply:
If a workaround is a huge pain we can require a more recent Bash version (which can easily be installed using `homebrew`).  But hopefully we can find a way that works on the old bash version.

Reply:
Believed fixed for macOS in 202e34367561483dbd2a13c23708ff5971a4b6f3, but untested by me.

Reply:
Ok, I will test it out later just to make sure